fontembed/embed_pdf.c

```diff
@@ -125,14 +126,8 @@ static EMB_PDF_FONTDESCR *emb_pdf_fd_new(const char *fontname,
   ret->fontname=ret->data+len;
   len+=strlen(fontname)+1;
   if (subset_tag) {
-    //Allocate new array and copy in data
-    char *newArray = new char[7];
-    memcpy(newArray, ret->fontname,1);
-    //Delete old array
-    delete [] ret->fontname;
-    //Swap pointers and new size
-    ret->fontname = newArray;
-    strncpy(ret->fontname,subset_tag,6);
+    //strncpy(ret->fontname,subset_tag,6);
+    snprintf(ret->fontname, 6, "%s", subset_tag);
```

Comment:
Simply this change (strncpy() -> snprintf()) eliminates the warning. Both calls do the same, but the snprintf() is without warning. Probably the developers of gcc simply did not came up to add this new warning also to snprintf(). For now it works. It is only a question for how long.
I have now manually applied this change as commit 5e1aa85f531. Thank you very much for investigating this.